File: android/app/src/main/java/com/smartexpense/android/domain/usecase/user/UpdateMeUseCase.java
package com.smartexpense.android.domain.usecase.user;

import com.smartexpense.android.data.remote.dto.request.UpdateProfileRequestDto;
import com.smartexpense.android.data.remote.dto.response.UserProfileResponseDto;
import com.smartexpense.android.domain.repository.ResultCallback;
import com.smartexpense.android.domain.repository.UserRepository;

public class UpdateMeUseCase {
    private final UserRepository repository;

    public UpdateMeUseCase(UserRepository repository) {
        this.repository = repository;
    }

    public void execute(String displayName, String avatarUrl, Double monthlyBudget, String themeColor, ResultCallback<UserProfileResponseDto> callback) {
        UpdateProfileRequestDto request = new UpdateProfileRequestDto(displayName, avatarUrl, monthlyBudget, themeColor);
        repository.updateMe(request, callback);
    }
}

File: android/app/src/main/kotlin/com/smartexpense/android/domain/usecase/user/UpdateMeUseCase.kt
package com.smartexpense.android.domain.usecase.user

import com.smartexpense.android.data.remote.dto.request.UpdateProfileRequestDto
import com.smartexpense.android.data.remote.dto.response.UserProfileResponseDto
import com.smartexpense.android.domain.repository.UserRepository

class UpdateMeUseCase(private val repository: UserRepository) {
    suspend fun execute(displayName: String?, avatarUrl: String?, monthlyBudget: Double?, themeColor: String?): Result<UserProfileResponseDto> {
        val request = UpdateProfileRequestDto(displayName, avatarUrl, monthlyBudget, themeColor)
        return repository.updateMe(request)
    }
}
